devlib/native/macosx_native.cpp

```diff
@@ -163,6 +166,258 @@ namespace macxutil {
+        CFRelease(session);
+
+        return context.result;
+    }
+
+
+    void handleChildProcess(const char * filename, int * pipe, int * stdinpipe)
+    {
+        // Close unused pipes
+        ::close(pipe[0]);
+        ::close(stdinpipe[1]);
+
+        qCDebug(macxutil::macxlog()) << "CHILD: EXIT -1";
+
+        // Assign pipes with std in/out
+        ::dup2(pipe[1], STDOUT_FILENO);
+        ::dup2(stdinpipe[0], STDIN_FILENO);
+
+        const char * cmd = "/usr/libexec/authopen";
+        QByteArray mode = QByteArray::number(O_RDWR);
+        ::execl(cmd, cmd, "-stdoutpipe", "-extauth", "-o", mode.data(), filename, NULL);
+
+        // Exit with error code -1 in case of execl error
+        ::exit(-1);
+    }
+
+    auto receiveFdFromChildProcess(const int & fromChildToParentPipe) -> int
+    {
+        const size_t bufSize = CMSG_SPACE(sizeof(int));
+        char buf[bufSize];
+        struct iovec io_vec[1];
+        io_vec[0].iov_base = buf;
+        io_vec[0].iov_len = bufSize;
+        const size_t cmsgSize = CMSG_SPACE(sizeof(int));
+        char cmsg[cmsgSize];
+
+        struct msghdr msg = {0};
+        msg.msg_iov = io_vec;
+        msg.msg_iovlen = 1;
+        msg.msg_control = cmsg;
+        msg.msg_controllen = cmsgSize;
+
+        ssize_t size;
+        do {
+            size = recvmsg(fromChildToParentPipe, &msg, 0);
+        } while (size == -1 && errno == EINTR);
+
+        qCDebug(macxutil::macxlog()) << "RECEIVED SIZE:" << size;
+
+        if (size > 0) {
+            struct cmsghdr * chdr = CMSG_FIRSTHDR(&msg);
+            if (chdr && chdr->cmsg_type == SCM_RIGHTS) {
+                qCDebug(macxutil::macxlog()) << "SCMRIGHTS";
+                return *((int *)(CMSG_DATA(chdr)));
+            } else {
+                qCDebug(macxutil::macxlog()) << "NOT SCMRIGHTS";
+                return -1;
+            }
+        }
+        return -2;
+    }
+
+    auto waitForChildProcessToFinish(const int & processId) -> int
+    {
+        pid_t wpid;
+        int status;
+
+        do {
+            wpid = ::waitpid(processId, &status, 0);
+        } while (wpid == -1 && errno == EINTR);
+
+        if (wpid == -1) {
+            qCDebug(macxutil::macxlog()) << "waitpid() failed executing authopen";
+            return -2;
+        }
+        if (WEXITSTATUS(status)) {
+            qCDebug(macxutil::macxlog()) << "authopen returned failure code" << WEXITSTATUS(status);
+            return -3;
+        }
+
+        return 0;
+    }
+
+
+    auto authOpenStorageDevice(const QByteArray & filename) -> int
+    {
+        QByteArray right = "sys.openfile.readwrite." + filename;
+        AuthorizationItem item = {right, 0, nullptr, 0};
+        AuthorizationRights rights = {1, &item};
+        AuthorizationFlags flags = kAuthorizationFlagInteractionAllowed | kAuthorizationFlagExtendRights
+                                   | kAuthorizationFlagPreAuthorize;
+        AuthorizationRef authRef;
```

Comment:
Could be wrapped as well calling `AuthorizationFree` in the deleter.

Reply:
Applied